### PR TITLE
Support heaviside ufunc.

### PR DESCRIPTION
**Pull request by mhvk**
As noted by @saimn in ~~https://github.com/astropy/astropy/issues/2933#issuecomment-290009552~~ https://github.com/astropy/astropy/pull/5835#issuecomment-290055672, numpy-dev now has a `heaviside` ufunc. This adds support. Set the milestone to 1.3.2, since ideally all tests pass also on numpy-dev with 1.3.

### CODE REVIEW

File: astropy/units/quantity_helper.py [hunk 1]
```diff
@@ -86,13 +86,8 @@ def get_converter(from_unit, to_unit):
     UFUNC_HELPERS[np.cbrt] = lambda f, unit: (
         [None], (unit ** Fraction(1, 3) if unit is not None
                  else dimensionless_unscaled))
-# ones_like was not private in numpy <= 1.6
```
Comment (mhvk) [inline astropy/units/quantity_helper.py]: We haven't supported numpy 1.6 for quite a while, so I removed this...

File: astropy/units/tests/test_quantity_ufuncs.py [hunk 2]
```diff
@@ -273,6 +273,23 @@ def test_reciprocal_array(self):
...
+    # heaviside only introduced in numpy 1.13
+    @pytest.mark.skipif("not hasattr(np, 'heaviside')")
+    def test_heaviside_scalar(self):
+        assert np.heaviside(0. * u.m) == 0.5 * u.dimensionless_unscaled
+        assert np.heaviside(0. * u.s, 0.3) == 0.3 * u.dimensionless_unscaled
+        assert np.heaviside(2. * u.J, 0.3) == 1. * u.dimensionless_unscaled
+
+    @pytest.mark.skipif("not hasattr(np, 'heaviside')")
+    def test_heaviside_array(self):
+        # Calculate cbrt on both sides since on Windows the cube root of 64
```
Comment (adrn) [inline astropy/units/tests/test_quantity_ufuncs.py]: did this comment get copied accidentally?

Comment (mhvk) [inline astropy/units/tests/test_quantity_ufuncs.py]: Yes! Will correct.

**Comment by mhvk** [comment]
Well, the tests with `numpy-dev` do start to fail, and @saimn noticed it. I think earlier there was some suggestion of having a `cron` job which checks for such failures and e-mails me, but since there are so many failures due to URL time-outs, I never set that up...

**Comment by saimn** [comment]
Thanks @mhvk ! (ref https://github.com/astropy/astropy/pull/5835#issuecomment-290055672 actually)

**Comment by bsipocz** [comment]
@mhvk - What we could do is to add a numpy-dev job to the main matrix without letting it fail, but only run it from the daily cron while leaving the current one to run only for PRs and pulls and letting it fail. That would send the travis fail e-mail to all the maintainers so higher chance we would notice that it went wrong.

**Comment by mhvk** [comment]
@bsipocz - yes, that might be a good idea; in that case, we could also disallow remote data, so we don't get false positives.

**Comment by bsipocz** [comment]
OK. I’ll open a PR for that soon.

**Comment by pllim** [comment]
I was going to suggest a bot that monitors Numpy change log, but bsipocz's suggestion seems more reasonable. :sweat_smile:

**Comment by mhvk** [comment]
OK, I finally got the numpy-dev build not to have URL time-outs. Since @adrn OK'd it (I addressed his comment), I'll merge. Thanks again, @saimn, for noticing this!

**Pull request merged**